test_runner/src/main/kotlin/ftl/gc/Utils.kt

```diff
@@ -1,9 +1,33 @@
 package ftl.gc
 
+import com.google.api.services.testing.model.AndroidInstrumentationTest
 import com.google.api.services.testing.model.ClientInfoDetail
+import com.google.api.services.testing.model.ManualSharding
+import com.google.api.services.testing.model.ShardingOption
+import com.google.api.services.testing.model.TestTargetsForShard
+import com.google.api.services.testing.model.UniformSharding
+import ftl.args.AndroidArgs
+import ftl.args.ShardChunks
 
 internal fun Map<String, String>.toClientInfoDetailList() = map { (key, value) ->
     ClientInfoDetail()
         .setKey(key)
         .setValue(value)
 }
+
+internal fun AndroidInstrumentationTest.setupTestTargets(args: AndroidArgs, testShards: ShardChunks) = apply {
+    if (args.disableSharding) {
+        testTargets = testShards.flatten()
+    } else {
+        shardingOption = ShardingOption().apply {
+            if (args.numUniformShards != null) {
+                testTargets = testShards.flatten()
+                uniformSharding = UniformSharding().setNumShards(args.numUniformShards)
```

Comment:
When number of `numUniformShards` is bigger then number of tests flank we get error while processing results
```
java.lang.IllegalStateException: response.testCases must not be null
	at ftl.reports.api.CreateTestExecutionDataKt.createTestExecutionData(CreateTestExecutionData.kt:34)
	at ftl.reports.api.CreateTestExecutionDataKt$createTestExecutionData$1.invokeSuspend(CreateTestExecutionData.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.ResumeModeKt.resumeUninterceptedMode(ResumeMode.kt:45)
	at kotlinx.coroutines.internal.ScopeCoroutine.afterCompletionInternal(Scopes.kt:32)
	at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:310)
	at kotlinx.coroutines.JobSupport.tryFinalizeFinishingState(JobSupport.kt:236)
	at kotlinx.coroutines.JobSupport.tryMakeCompletingSlowPath(JobSupport.kt:849)
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:811)
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:787)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:111)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)
```

In this particular case test apk had 7 tests and `numUniformShards` was set to `10`.  As result 10 shards was launched.

Reply:
Fixed